pkg/cluster/util.go

```diff
@@ -517,36 +517,43 @@ func (c *Cluster) dnsName(role PostgresRole) string {
 	// if cluster name starts with teamID we might need to provide backwards compatibility
 	clusterNameWithoutTeamPrefix, _ := acidv1.ExtractClusterName(c.Name, c.Spec.TeamID)
 	if clusterNameWithoutTeamPrefix != "" {
-		if role == Replica {
-			clusterNameWithoutTeamPrefix = fmt.Sprintf("%s-repl", clusterNameWithoutTeamPrefix)
+		if role == Master {
+			oldDnsString = c.oldMasterDNSName(clusterNameWithoutTeamPrefix)
+		} else {
+			oldDnsString = c.oldReplicaDNSName(clusterNameWithoutTeamPrefix)
 		}
-		dnsString = fmt.Sprintf("%s,%s", dnsString, c.oldDNSFormat(clusterNameWithoutTeamPrefix))
+		dnsString = fmt.Sprintf("%s,%s", dnsString, oldDnsString)
 	}
 
 	return dnsString
 }
 
 func (c *Cluster) masterDNSName() string {
-	return strings.ToLower(c.OpConfig.MasterDNSNameFormat.Format(
+	return strings.ToLower(c.OpConfig.MasterLBDNSNameFormat.Format(
 		"cluster", c.Name,
 		"namespace", c.Namespace,
-		"team", c.teamName(),
 		"hostedzone", c.OpConfig.DbHostedZone))
```

Comment:
docs say `No other placeholders are allowed!` so we should not have `"team", c.teamName()`. Team for a cluster can change. Should not be part of the DNS anyway.